src/coreclr/vm/methodtable.cpp

```diff
@@ -4220,19 +4221,31 @@ OBJECTREF MethodTable::GetManagedClassObject()
         // Make sure that we have been restored
         CheckRestore();
 
-        REFLECTCLASSBASEREF  refClass = NULL;
+        REFLECTCLASSBASEREF refClass = NULL;
         GCPROTECT_BEGIN(refClass);
-        refClass = (REFLECTCLASSBASEREF) AllocateObject(g_pRuntimeTypeClass);
 
-        LoaderAllocator *pLoaderAllocator = GetLoaderAllocator();
-
-        ((ReflectClassBaseObject*)OBJECTREFToObject(refClass))->SetType(TypeHandle(this));
-        ((ReflectClassBaseObject*)OBJECTREFToObject(refClass))->SetKeepAlive(pLoaderAllocator->GetExposedObject());
+        LOADERHANDLE exposedClassObjectHandle = NULL;
```

Comment:
This is not a technically `LOADERHANDLE` since it does not follow the `LOADERHANDLE` contract anymore.

Use a different name (e.g. like `RUNTIMETYPEHANDLE`) to avoid confusion?